src/main/java/io/jenkins/docker/connector/DockerComputerConnector.java

```diff
@@ -108,16 +109,35 @@ public void afterContainerStarted(@Nonnull DockerAPI api, @Nonnull String workdi
     /**
      * Ensure container is already set with a command, or set one to make it wait
      * indefinitely
-     * 
+     *
+     * @param workdir
      * @param cmd The {@link CreateContainerCmd} to be adjusted.
      */
-    protected void ensureWaiting(@Nonnull CreateContainerCmd cmd) {
+    protected void ensureWaiting(String workdir, @Nonnull CreateContainerCmd cmd) {
         final String[] cmdAlreadySet = cmd.getCmd();
+
         if (cmdAlreadySet == null || cmdAlreadySet.length == 0) {
-            // no command has been set, we need one that will just hang. Typically "sh" waiting for stdin
-            cmd.withCmd("/bin/sh")
-               .withTty(true)
-               .withAttachStdin(false);
+            // on windows we can't rely on /bin/sh being present, so we use cmd.exe to ensure a process is waiting
+            // TODO: same here, refined detection mechanism if possible
+            if(workdir.matches("^.:.*")) {
```

Comment:
The CMD field can be set in the template definition.
While it might be nice to have the code set a Windows-compatible default, it's not essential as the user can specify one themselves.
i.e. this can be addressed in the help text.